DeepDiveLogCompiler.scala

```diff
@@ -425,13 +491,33 @@ object DeepDiveLogCompiler extends DeepDiveLogHandler {
     // take an initial pass to analyze the parsed program
     val state = new CompilationState( programToCompile )
 
+    // compile compilation states by head name based on type
```

Comment:
There are near identical lines of code in `CompilationState.analyzeDependency` as below.  Can we move these into CompilationState class and reuse what's been analyzed already from here?  Generally, we should keep the `run` super short and simple, and push the details down to other functions.